MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java

```diff
@@ -0,0 +1,290 @@
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconAllowOverlap;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconIgnorePlacement;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconImage;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconOffset;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconTranslate;
+
+/**
+ * Combine SymbolLayer icons with the Android system's ValueAnimator and interpolator
+ * animation for a fun pin drop effect. The interpolator movement can also be used with other
+ * types of map layers, such as a LineLayer or CircleLayer.
+ * <p>
+ * More info about https://developer.android.com/reference/android/view/animation/Interpolator
+ */
+public class ValueAnimatorIconAnimationActivity extends AppCompatActivity implements
+  OnMapReadyCallback, MapView.OnDidFinishRenderingMapListener {
+
+  private static final String ICON_ID = "red-pin-icon-id";
+
+  // This float's actual value will depend on the height of the SymbolLayer icon
+  private static final float DEFAULT_DESIRED_ICON_OFFSET = -16;
+  private static final float STARTING_DROP_HEIGHT = -100;
+  private static final long DROP_SPEED_MILLISECONDS = 1200;
+  private MapView mapView;
+  private SymbolLayer pinSymbolLayer;
+  private Style style;
+  private TimeInterpolator currentSelectedTimeInterpolator;
+  private ValueAnimator animator;
+  private boolean firstRunThrough;
+  private boolean animationHasStarted;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    currentSelectedTimeInterpolator = new BounceInterpolator();
+    firstRunThrough = true;
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_animated_pin_drop);
+
+    // Initialize the map view
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+    mapboxMap.setStyle(Style.LIGHT, new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        ValueAnimatorIconAnimationActivity.this.style = style;
+        mapView.addOnDidFinishRenderingMapListener(ValueAnimatorIconAnimationActivity.this);
+        initLayerIcon(style);
+        initDataSource(style);
```

Comment:
nit can be added with the StyleBuilder#withSource/Layer

Reply:
Added source via StyleBuilder, but keeping `initSymbolLayer()` as is if it's ok. The example is so fine-grained with FABS, etc. that it's easier to initialize the layer a bit after the style is actually created..

Reply:
<img width="663" alt="Screen Shot 2019-03-29 at 2 16 46 PM" src="https://user-images.githubusercontent.com/4394910/55263151-4ec43c00-522d-11e9-8e96-588ca6b3d2db.png">
<img width="531" alt="Screen Shot 2019-03-29 at 2 16 42 PM" src="https://user-images.githubusercontent.com/4394910/55263153-4ec43c00-522d-11e9-8070-218dcadbb80e.png">